spec/features/publish/course_confirmation_spec.rb

```diff
@@ -0,0 +1,65 @@
+require "rails_helper"
+
+feature "course confirmation" do
+  before do
+    given_i_am_authenticated_as_a_provider_user
+    when_i_visit_the_confirmation_page
+  end
+
+  scenario "creating course" do
+    and_i_click_continue
+
+    then_i_am_met_with_the_publish_provider_courses_index_page
+  end
+
+  scenario "it displays the correct information" do
+    then_it_displays_correctly
+  end
+
+private
+
+  def given_i_am_authenticated_as_a_provider_user
+    @user = create(:user, providers: [build(:provider, sites: [build(:site)])])
+    @user.providers.first.courses << create(:course, :with_accrediting_provider)
+    given_i_am_authenticated(user: @user)
+  end
+
+  def when_i_visit_the_confirmation_page
+    confirmation_page.load(provider_code: provider.provider_code, recruitment_cycle_year: Settings.current_recruitment_cycle_year, query: confirmation_params(provider))
```

Comment:
could we break this up, bit long

Reply:
done in https://github.com/DFE-Digital/teacher-training-api/pull/2525/commits/77827debc4ed2be38058f60183926ff5264ee34a